src/migration/1687960180394-PostFlags.ts

```diff
@@ -5,17 +5,36 @@ export class PostFlags1687960180394 implements MigrationInterface {
 
     public async up(queryRunner: QueryRunner): Promise<void> {
         await queryRunner.query(`ALTER TABLE "post" ADD "flags" jsonb NOT NULL DEFAULT '{}'`);
+
         await queryRunner.query(`CREATE INDEX "IDX_post_flags_sentAnalyticsReport" ON post USING HASH (((flags->'sentAnalyticsReport')::boolean))`);
         await queryRunner.query(`CREATE INDEX "IDX_post_flags_banned" ON post USING HASH (((flags->'banned')::boolean))`);
         await queryRunner.query(`CREATE INDEX "IDX_post_flags_deleted" ON post USING HASH (((flags->'deleted')::boolean))`);
         await queryRunner.query(`CREATE INDEX "IDX_post_flags_promoteToPublic" ON post USING HASH (((flags->'promoteToPublic')::boolean))`);
+
+        await queryRunner.query(`
+          CREATE OR REPLACE FUNCTION post_default_flags()
+            RETURNS TRIGGER
+            LANGUAGE PLPGSQL
+            AS
+          $$
+          BEGIN
+            NEW."flags" = '{"sentAnalyticsReport":true,"banned":false,"deleted":false,"private":false,"visible":true,"showOnFeed":true,"promoteToPublic":false}' || NEW."flags";
+            RETURN NEW;
+          END;
+          $$
+        `)
+        await queryRunner.query('CREATE TRIGGER post_default_flags_trigger BEFORE INSERT ON "post" FOR EACH ROW EXECUTE PROCEDURE post_default_flags()')
```

Comment:
Added so that before insert we merge all the flags provided by insert statement with defaults.

This also takes care of the case where in graphql if we query for example `flags.private` in our `Post` query (as seen in `should return default values for public flags` test) it would return:
```json
{
 "flags": {
    "private": null
  }
}
```

Which I think it is not really good from client side. Granted we could take care of this with graphorm transform but then querying from db directly would again have null values. 

Since currently we had all the fields populated by default for each of the columns I think this should follow case.

This continues on points from https://github.com/dailydotdev/daily-api/pull/1323#discussion_r1253482759